### PR TITLE
[docker_daemon] Add metadata, tests and fix missing yaml option

### PR DESCRIPTION
**Pull request by hkaj**
# Why this PR
This PR is a fix/improvement for https://github.com/DataDog/dd-agent/pull/2405
It adds the same metrics for metadata and modify the existing name for consistency (this has not been released so no user will be impacted), setup testing and fix a missing option in `docker_daemon.yaml`.

### CODE REVIEW

File: checks.d/docker_daemon.py [hunk 1]
```diff
@@ -684,6 +697,24 @@ def _format_disk_metrics(self, metrics):
...
+            used = stats.get('docker.{0}.used'.format(mtype))
+            total = stats.get('docker.{0}.total'.format(mtype))
+            free = stats.get('docker.{0}.free'.format(mtype))
+            if total < free + used:
+                self.log.error('used, free, and total disk metrics may be wrong, '
+                               'unable to calculate percentage.')
+                return {}
+            if used > 0.0 and total > 0.0:
+                percs['docker.{0}.percent'.format(mtype)] = round(100 * float(used) / float(total), 2)
+            elif free > 0.0 and total > 0.0:
```
Comment (truthbk) [inline checks.d/docker_daemon.py]: I think `free` could legitly be 0.0.

File: checks.d/docker_daemon.py [hunk 2]
```diff
@@ -684,6 +697,24 @@ def _format_disk_metrics(self, metrics):
...
+        percs = {}
+        for mtype in mtypes:
+            used = stats.get('docker.{0}.used'.format(mtype))
+            total = stats.get('docker.{0}.total'.format(mtype))
+            free = stats.get('docker.{0}.free'.format(mtype))
+            if total < free + used:
+                self.log.error('used, free, and total disk metrics may be wrong, '
+                               'unable to calculate percentage.')
+                return {}
+            if used > 0.0 and total > 0.0:
```
Comment (truthbk) [inline checks.d/docker_daemon.py]: Might be worth looking at handling total = 0.0 simply as an exception. Not sure what might look cleaner.

File: tests/checks/mock/test_docker_daemon.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+    def mock_get_info_no_used(self):
+        return {
+            'DriverStatus': [
+                ['Data Space Available', '9 GB'],
+                ['Data Space Total', '10 GB'],
+                ['Metadata Space Available', '9 MB'],
+                ['Metadata Space Total', '10 MB'],
+            ]
+        }
```
Comment (truthbk) [inline tests/checks/mock/test_docker_daemon.py]: We probably want to add a couple more cases with edge cases like total = 0 GB, free = 0 Gb.

**Comment by hkaj** [comment]
Thanks for the feedback @truthbk, updated some things and added some testing.

File: checks.d/docker_daemon.py [hunk 4]
```diff
@@ -682,8 +699,31 @@ def _format_disk_metrics(self, metrics):
...
 
+    def _calc_percent_disk_stats(self, stats):
+        """Calculate a percentage of used disk space for data and metadata"""
+        mtypes = ['data', 'metadata']
+        percs = {}
+        for mtype in mtypes:
+            used = stats.get('docker.{0}.used'.format(mtype))
+            total = stats.get('docker.{0}.total'.format(mtype))
+            free = stats.get('docker.{0}.free'.format(mtype))
+            if used and total and free and total < free + used:
```
Comment (remh) [inline checks.d/docker_daemon.py]: Should we add a small rounding tolerance factor here ?

**Comment by truthbk** [comment]
I'd say that other than @remh consideration, it's looking good. With regards to his point, maybe it makes sense to use the ceiling of "total"? I'll let you decide on that one :)

**Comment by hkaj** [comment]
@truthbk added a ceiling of total as a rounding margin.

**Comment by truthbk** [comment]
👍  looks good... I was unable to run the tests locally. But I trust they're fine after the revamp. Feel free to squash and merge.
